lib/wdioUi5-index.js

```diff
@@ -244,29 +252,75 @@ function setup(context) {
     });
 
     /**
-     * this function is the main method to interact with the UI5 application
+     * creates a string valid as object key from a selector
+     * @param {sap.ui.test.RecordReplay.ControlSelector} selector
+     * @returns {String} wdio_ui5_key
+     */
+    function createWdioUI5KeyFromSelector(selector) {
+
+        var orEmpty = (string) => {
+            return string || "-";
+        };
+        const wdi5_ui5_key = stripNonValidCharactersForKey(`${orEmpty(selector.id)}_${orEmpty(selector.viewName)}_${orEmpty(selector.controlType)}_${orEmpty(selector.bindingPath)}_${orEmpty(selector.I18NText)}_${orEmpty(selector.labelFor)}_${orEmpty(selector.properties)}`);
+        return wdi5_ui5_key;
+    };
+
+    /**
+     * to generate an object key from any string
+     * @param {String} key
+     * @returns {String}
+     */
+    function stripNonValidCharactersForKey(key) {
+        key = key.split('.').join('')
+            .split('/').join('')
+            .split(' ').join('')
+            .split('>').join('')
+            .split('_-').join('')
+            .split('-').join('')
+            .toLowerCase();
+        return key;
+    };
+
+    async function createWdi5Control(wdi5Selector, context) {
+
+        // create WDI5 control
+        const wdi5Control = new WDI5(wdi5Selector, context);
+
+        // save control
+        context._controls[wdi5Selector.wdio_ui5_key] = wdi5Control;
+        logger.info(`creating internal control with id ${wdi5Selector.wdio_ui5_key}`);
+
+        return wdi5Control;
+    };
+
+    /**
+     * this function is the main method to enable the communication with the UI5 application
      * can be accessed via the browser object in the test case `browser.asControl(selector)` whereas the selector is of type WDI5Selector
+     *
+     * wdio_ui5_key internally used key to store the already retrieved controls and prevent double browser access.
+     * Check for object type, if wdio_ui5_key is present.
+     *
+     * const wdi5Selector = {
+     *     wdio_ui5_key: „someCutomControlIdentifier“,
+     *     selector: <sap.ui.test.RecordReplay.ControlSelector>
+     * }
+     *
+     * wdio_ui5_key is generated based on the givven selector.
+     * If wdio_ui5_key is provided with the selector the provided wdio_ui5_key is used.
+     * You can force to load the control freshly from browser context by setting the 'forceSelect' parameter to true
+     *
      * @param {WDI5Selector} wdi5Selector custom selector object with property wdio_ui5_key and sap.ui.test.RecordReplay.ControlSelector
      */
-    context.addCommand('asControl', (wdi5Selector) => {
-        // facade on a UI5 control
-        // to expose certain methods of the UI5 control via wdio
+    context.addCommand('asControl', async (wdi5Selector) => {
```

Comment:
why flagged as `async` here? there doesn't seem to be actual async (`await` or such) code in here